mod/profile.php

```diff
@@ -150,42 +150,17 @@ function profile_content(App $a, $update = 0)
 		Nav::setSelected('home');
 	}
 
-	$contact = null;
-	$remote_contact = false;
-
-	$contact_id = 0;
-
-	if (!empty($_SESSION['remote'])) {
-		foreach ($_SESSION['remote'] as $v) {
-			if ($v['uid'] == $a->profile['profile_uid']) {
-				$contact_id = $v['cid'];
-				break;
-			}
-		}
-	}
-
-	if ($contact_id) {
-		$groups = Group::getIdsByContactId($contact_id);
-		$r = q("SELECT * FROM `contact` WHERE `id` = %d AND `uid` = %d LIMIT 1",
-			intval($contact_id),
-			intval($a->profile['profile_uid'])
-		);
-		if (DBA::isResult($r)) {
-			$contact = $r[0];
-			$remote_contact = true;
-		}
-	}
+	$remote_contact = Contact::isFollower(remote_user(), $a->profile['profile_uid']);
+	$is_owner = local_user() == $a->profile['profile_uid'];
+	$last_updated_key = "profile:" . $a->profile['profile_uid'] . ":" . local_user() . ":" . remote_user();
 
-	if (!$remote_contact) {
-		if (local_user()) {
-			$contact_id = $_SESSION['cid'];
-			$contact = $a->contact;
+	if ($remote_contact) {
+		$cdata = Contact::getPublicAndUserContacID(remote_user(), $a->profile['profile_uid']);
+		if (!empty($cdata['user'])) {
```

Comment:
This is already checked in `Contact::isFollower` on line 153.

Reply:
I need the user's contact id to fetch the associated groups this contact belongs to. I'm not totally glad of doing repeated database calls, but I guess we would had to completely redesign the "display" functionality to really get rid of this.

Reply:
I thought the underlying `Contact::getIdForURL` was cached but it doesn't seem to be the case.

Reply:
The whole stuff is ugly, but since it isn't called in the regular daily (local) business, there is not that big performance drain here.